From cc80812c01473c4b50131614f836a77d5ab8256c Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Wed, 29 Jun 2022 17:35:28 +0200
Subject: [PATCH 207/954] fix(stm32mp2): DDR FW management via FIP

Update DDR firmware load address in SRAM1 after the reserved memory for
BSEC shadow to avoid conflict.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git ce926302a3a745a18b2b4df4693ca996e1bcde59 fix(stm32mp2): DDR FW management via FIP]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Id49dc858a9ab6c186bb88101dd08b6c94822de9f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/258456
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index 8e4bd1bd9f..e485e815a8 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -175,7 +175,7 @@ enum ddr_type {
 #endif
 
 #if STM32MP_DDR_FIP_IO_STORAGE
-#define STM32MP_DDR_FW_BASE		SRAM1_BASE
+#define STM32MP_DDR_FW_BASE		(SRAM1_BASE + STM32MP_SEC_SRAM1_SIZE)
 #define STM32MP_DDR_FW_DMEM_OFFSET	U(0x400)
 #define STM32MP_DDR_FW_IMEM_OFFSET	U(0x800)
 #define STM32MP_DDR_FW_MAX_SIZE		U(0x8800)
-- 
2.53.0

